### PR TITLE
APIGOV-26363 - fix several issues related to proxy mode discovery

### PR DESCRIPTION
**Pull request by jcollins-axway**
Fixes 

- #140 
- #142 
- #143 
- #146

### CODE REVIEW

File: client/pkg/apigee/proxy.go [hunk 1]
```diff
@@ -65,6 +82,50 @@ func (a *ApigeeClient) GetRevision(proxyName, revision string) (*models.ApiProxy
...
+func (a *ApigeeClient) GetRevisionConnectionType(proxyName, revision string) (*HTTPProxyConnection, error) {
+	response, err := a.newRequest(http.MethodGet, fmt.Sprintf("%s/apis/%s/revisions/%s", a.orgURL, proxyName, revision),
+		WithDefaultHeaders(),
+		WithQueryParam("format", "bundle"),
+	).Execute()
+	if err != nil {
+		return nil, err
+	}
+
+	// response is a zip file, lets
```
Comment (sbolosan) [inline client/pkg/apigee/proxy.go:95]: comment is dangling here

File: client/pkg/apigee/proxy.go [hunk 2]
```diff
@@ -65,6 +82,50 @@ func (a *ApigeeClient) GetRevision(proxyName, revision string) (*models.ApiProxy
...
+		if zipFile.Name != "apiproxy/proxies/default.xml" {
+			continue
+		}
+		fileBytes, err = readZipFile(zipFile)
+		if err != nil {
+			return nil, err
+		}
+		break
+	}
+
```
Comment (sbolosan) [inline client/pkg/apigee/proxy.go:113]: do you wanna check in case none of the files were "apiproxy/proxies/default.xml"

Comment (jcollins-axway) [reply client/pkg/apigee/proxy.go:113]: good call, I will return an error if the file is not found

**Review by sbolosan — APPROVED** [review]
solid...

**Review by vivekschauhan — APPROVED** [review]

**Pull request merged**